users/vieting/jobs/fairseq_finetuny.py

```diff
@@ -0,0 +1,185 @@
+import soundfile
+import glob
+import os
+import random
+import subprocess
+import os
+
+from i6_core.lib import corpus
+
+from sisyphus import Job, Task, tk
+
+
+class CreateTaskDataJob(Job):
+    """
+    Creates required task files for wav2vec finetuning with fairseq. This includes the following files:
+    - train.tsv
+    - train.ltr
+    - train.wrd
+    - valid.tsv
+    - valid.ltr
+    - valid.wrd
+    - dict.ltr.txt
+
+    For the script see https://github.com/pytorch/fairseq/blob/main/examples/wav2vec/wav2vec_manifest.py for .tsv creation,
+    https://github.com/facebookresearch/fairseq/blob/91c364b7ceef8032099363cb10ba19a85b050c1c/examples/wav2vec/libri_labels.py aswell as
+    the issue https://github.com/facebookresearch/fairseq/issues/2493 for .wrd and .ltr creation.
+    """
+
+    def __init__(
+        self,
+        corpus_paths,
+        file_extension="wav",
+        valid_percent=0.01,
+        seed=42,
+        path_must_contain=None
+    ):
+        """
+        :param [tk.Path]|tk.Path audio_dir_path: list of paths or single path to raw audio files to be included
+        :param str file_extension: file extension to look for in audio_dir_path
+        :param float valid_percent: percentage of files to be in validation set
+        :param int seed: random seed for splitting into train and valid set
+        :param str|None path_must_contain: if set, path must contain this substring
+            for a file to be included in the task
+        """
+        if isinstance(corpus_paths, tk.Path):
+            self.corpus_paths = [corpus_paths]
+        else:
+            assert isinstance(corpus_paths, list)
+            self.corpus_paths = corpus_paths
+        assert min([isinstance(path, tk.Path) for path in self.corpus_paths])
+
+        self.file_extension = file_extension
+        self.valid_percent = valid_percent
+        assert 0.0 <= self.valid_percent <= 1.0
+        self.seed = seed
+        self.path_must_contain = path_must_contain
+
+        self.rqmt = {"time": 6, "mem": 8, "cpu": 1}
+
+        self.out_task_path = self.output_path("task", directory=True)
```

Comment:
This seems to be unused.

Reply:
Regarding `self.out_task_path`: See comment above for similar line
Regarding `self.rqmt`: Should I completely remove this? I just copied the `rqmt` from the manifest creation job, assuming that this is necessary for every job class

Reply:
Yes, see comment above.

`self.rqmt` should stay. It's also used in `self.tasks()`. I just think it could be reduced, right? You mentioned the job takes 5-10 minutes, so `"time": 1` should be more than enough. Regarding memory, I'm not sure how much you'd need there, but probably also much less. You can give it a try, but I don't think it's too important to "tune" this.

Reply:
I just tested to reduce the memory, however then the manager somehow stopped logging after some time, leading to some weird error messages, I would just leave that then as it is.
Regarding the `time` requirement, the 5-10mins was only when creating labeled data from a single corpus. However, when training with LS960 for example, we have to create labeled data from 3 corpus files, leading to a much longer execution time. I would probably just leave this as it is as well.